grader/src/main/java/edu/pdx/cs410J/grader/scoring/ProjectSubmissionOutputFileParser.java

```diff
@@ -1,27 +1,106 @@
 package edu.pdx.cs410J.grader.scoring;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.IOException;
 import java.io.Reader;
+import java.io.StringReader;
+import java.nio.CharBuffer;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.time.format.DateTimeFormatter;
 import java.util.Date;
+import java.util.Locale;
+import java.util.regex.Pattern;
 
 /**
  * Author: Susham Kumar
  */
 
 public class ProjectSubmissionOutputFileParser {
-  public ProjectSubmissionOutputFileParser(Reader source) {
 
+  public ProjectSubmissionOutputFileParser(Reader source) {
+      if(source != null) {
+          bufferedReader = new BufferedReader(source);
+      }
 
   }
 
-  public ProjectSubmission parse() {
-    throw new UnsupportedOperationException("Not implemented yet");
+  public ProjectSubmission parse()  {
+
+     projectSubmission = new ProjectSubmission();
+     String fileContents=null;
+
+      try {
+          while((fileContents= bufferedReader.readLine()) !=null) {
+            fileContents=fileContents.trim();
+          if(fileContents.contains("CS410J")){              // fileContents has the student Id and Project Name.
+              String[] splitContent=fileContents.split(Pattern.quote("."));
+              projectSubmission.setProjectName(splitContent[splitContent.length -1]);
+              projectSubmission.setStudentId(splitContent[splitContent.length -2]);
+
+            }
+
+            if(fileContents.contains("Submitted by")){ //fileContents has StudentName
```

Comment:
I recommend that you move "Submitted by" into a constant (a private static final field whose name is ALL_CAPS) so that this "magic" string doesn't appear twice.

Reply:
Do you think, its better to create final fields whose names are ALL_CAPS, so that the magic string doesn't appear twice?

Reply:
Yes.  That way, if the "magic" string ever changes in the output file, we only need to change one place in the code.